### PR TITLE
Fill out Wolf implementation

### PR DESCRIPTION
**Pull request by Aaron1011**
I've also implemented `EntityTameable`, and added the necessary `getName` to `EntityPlayer` so that it can be used as a `Tamer`

### CODE REVIEW

**Comment by Minecrell** [comment]
If you add `getName()` to `EntityPlayer`, you should be able to remove `getName()` from EntityPlayerMP.

File: src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java [hunk 1]
```diff
@@ -71,4 +75,8 @@ public MixinEntityPlayer(World worldIn) {
         return this.openContainer == this.inventoryContainer;
     }
 
+    public String human$getName() {
+        return this.getName();
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java]: Like https://github.com/SpongePowered/Sponge/commit/07d73f4b74f7851bf23b81bd01a4998a6d9df8fc#commitcomment-9648317, this will cause an infinite loop in the deobfuscated environment because `this.getName()` and `human$getName()` will be the same. You need to implement it like https://github.com/SpongePowered/Sponge/commit/07154247ec9fe21e81117a43a2326de0e20c3bf7

Comment (Aaron1011) [inline src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java]: Oops, I thought I had added that. I'll fix it.

**Comment by Zidane** [comment]
@Aaron1011 

Fix conflicts.

File: src/main/java/org/spongepowered/mod/mixin/entity/living/MixinEntityTameable.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    @Overwrite
+    public Entity getOwner() {
+        return this.func_180492_cm();
+    }
+
+    public void sittable$setOwner(Tamer tamer) {
+        if (tamer instanceof EntityPlayer) {
+            this.func_152115_b(((EntityPlayer) tamer).getName());
```
Comment (simon816) [inline src/main/java/org/spongepowered/mod/mixin/entity/living/MixinEntityTameable.java]: Should be a UUID string not a name. See the usage of `func_152115_b` in vanilla

File: src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java (hunk 1 above)
Comment (simon816) [inline src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java]: A new way of fixing this error has just been pushed to the Mixin repo, you can use an [`@Intrinsic`](https://github.com/SpongePowered/Mixin/blob/master/src/main/java/org/spongepowered/asm/mixin/Intrinsic.java) annotation.

Comment (Mumfrey) [inline src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java]: Actually just for the record, the way @Aaron1011 has done this is exactly correct. By placing the `@Overwrite` with the original code *below* the interface accessor, it will *not* cause the infinite loop. However @simon816 is absolutely correct, this can now be solved with [`@Intrinsic`](https://github.com/SpongePowered/Mixin/blob/master/src/main/java/org/spongepowered/asm/mixin/Intrinsic.java) because the purpose of `Intrinsic` is precisely to avoid this kind of unnecessary duplication.

So basically, this code is quite correct - per my examples - but using `Intrinsic` will make it better.

Comment (Aaron1011) [inline src/main/java/org/spongepowered/mod/mixin/entity/player/MixinEntityPlayer.java]: I'm about to push code that changes it - I'm just debugging some other issues first.

File: src/main/java/org/spongepowered/mod/mixin/entity/living/MixinEntityTameable.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    @Shadow
+    public abstract void func_175544_ck();
+
+    @Shadow
+    public abstract EntityLivingBase func_180492_cm();
+
+    @Shadow
+    public abstract void func_152115_b(String name);
+
+    private Tamer tamer;
```
Comment (simon816) [inline src/main/java/org/spongepowered/mod/mixin/entity/living/MixinEntityTameable.java]: This field will not persist over server restarts, see https://github.com/SpongePowered/Sponge/pull/173 for how to fix the issue.

**Comment by Aaron1011** [comment]
This is blocked on (more of) the Data API being implemented.
